packages/amazonq/test/e2e/amazonq/featureDev.test.ts

```diff
@@ -283,15 +175,45 @@ describe('Amazon Q Feature Dev', function () {
             )
         })
 
-        it('Clicks examples', async () => {
-            const q = framework.createTab()
-            q.addChatMessage({ command: '/dev' })
-            q.clickButton(FollowUpTypes.DevExamples)
+        afterEach(async function () {
+            // currentTest.state is undefined if a beforeEach fails
+            if (
+                this.currentTest?.state === undefined ||
+                this.currentTest?.isFailed() ||
+                this.currentTest?.isPending()
+            ) {
+                // Since the tests are long running this may help in diagnosing the issue
+                console.log('Current chat items at failure')
+                console.log(JSON.stringify(tab.getChatItems(), undefined, 4))
+            }
+        })
 
-            const lastChatItems = q.getChatItems().pop()
-            assert.deepStrictEqual(lastChatItems?.body, examples)
+        it('Clicks accept code and click new task', async () => {
+            await retryIfRequired(async () => {
+                await Promise.any([
+                    waitForButtons([FollowUpTypes.InsertCode, FollowUpTypes.ProvideFeedbackAndRegenerateCode]),
+                    waitForButtons([FollowUpTypes.Retry]),
+                ])
+            })
+            tab.clickButton(FollowUpTypes.InsertCode)
+            await waitForButtons([FollowUpTypes.NewTask, FollowUpTypes.CloseSession])
+            tab.clickButton(FollowUpTypes.NewTask)
+            await waitForText('What new task would you like to work on?')
+            assert.deepStrictEqual(tab.getChatItems().pop()?.body, 'What new task would you like to work on?')
         })
 
-        functionalTests()
+        it('Iterates on codegen', async () => {
+            await retryIfRequired(async () => {
+                await Promise.any([
+                    waitForButtons([FollowUpTypes.InsertCode, FollowUpTypes.ProvideFeedbackAndRegenerateCode]),
+                    waitForButtons([FollowUpTypes.Retry]),
```

Comment:
Did not see where the retry button being used.

Reply:
basically this test path is the following:

wait until either one of the scenarios:
- retry button exists
- insert code & provide feedback button exists

if there is retry button then click retry, otherwise do nothing

retry button is used in the function `retryIfRequired`